app/Jobs/ProcessNewVideo.php

```diff
@@ -251,24 +253,81 @@ protected function generateVideoThumbnail($path, $time, $width, $height, $format
      */
     protected function getThumbnailTimes($duration)
```

Comment:
Why modify this function, as it is not used by your new code?

Reply:
When the duration is 0s, `$times` has a length of 10 in:
```php 
$times = $this->getThumbnailTimes($this->video->duration);
```
This situation causes the test function `testGenerateSpritesZeroDuration` to fail. Nevertheless, it can be resolved using the same logic as the other new test functions.